Snail/Observable.swift

```diff
@@ -61,7 +61,7 @@ public class Observable<T>: ObservableType {
 
     public func on(_ queue: DispatchQueue) -> Observable<T> {
         let observable = Observable<T>()
-        subscribe(queue: queue,
+        _ = subscribe(queue: queue,
                   onNext: { observable.on(.next($0)) },
                   onError: { observable.on(.error($0)) },
                   onDone: { observable.on(.done) })
```

Comment:
I think we should add this to the subscribers array right? This way we remove this subscription when `deinit`